test/src/common/bcmp_dfu_ut.cpp

```diff
@@ -118,7 +118,7 @@ class BcmpDfuTest : public ::testing::Test {
         .hwVersion = 1,
         .flags = 0,
         .versionStrLen = 5,
-        .versionStr = {"test"},
```

Comment:
This is apparently a C99 bug that was causing this test to fail. The suggested fix was found here:
https://pcbartists.com/firmware/designator-outside-aggregate-initializer-solved/

Reply:
@towynlin I would double check this runs fine on Mac. I have seen silly things like this break across Linux->Mac and visa-versa.I see our CI uses Ubuntu. Let me know!

Reply:
It does in fact fail to build on my mac arm machine:
```
bm_protocol/test/src/common/sensorSampler_ut.cpp:99:17: error: braces around scalar initializer [-Werror,-Wbraced-scalar-init]
      node[i] = SENSOR_NODE_INIT(testInit, testPoll, testCheck);
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
bm_protocol/src/lib/sensor_sampler/sensorSampler.h:21:20: note: expanded from macro 'SENSOR_NODE_INIT'
          .timer = {0}, \
                   ^~~
```
I'm looking into it.

Reply:
I'm noticing now that I see a different failure than you called out here. Sorry! I'll clarify once I figure that out.

Reply:
So yeah, thanks for calling out the need to test. It passes on my machine. That said, I see we don't have any tests that exercise whether `versionStr` is set as expected. Could you add that expectation to at least one of the tests?

Reply:
Will do!

Reply:
I added an `EXPECT_STREQ` for `versionStr` to ensure it is properly set during `SetUp` of the tests.